homework-submission/week1/src/components/TodoItem.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import ItemDeadLine from './ItemDeadLine.js';
+import CheckBox from './CheckBox.js';
+import { observer, inject } from 'mobx-react';
+
+@inject('todos')
+@observer
+class TodoItem extends React.Component {
+  render() {
+    const todo = this.props.todo;
+    const { editing } = this.props.todos;
+    const edit = editing === todo._id;
+    const content =
+      edit ? <input type='text' defaultValue={todo.description} ref={(input) => this.newDescription = input} />
+        : todo.description;
+    return (
+      <li className='todoItem'>
+        <div>
+          <CheckBox todo={todo} />
+          <label className={todo.done ? 'checked' : 'notChecked'}>{content}
+            <ItemDeadLine deadline={todo.deadline} />
+          </label>
+        </div>
+        {edit ?
+          <React.Fragment>
+            <button onClick={() => this.props.todos.updateHandler(todo._id, this.newDescription.value)} id={todo._id}>Update</button>
+            <button onClick={this.props.todos.editHandler} >Cancel</button>
```

Comment:
this is a good example of legibility, this is what I see plus many lines just like it will make it a bit confusing:

![screenshot_2018-09-12 husam - homework week 3 by threepalmtrees pull request 217 hackyourfuture react](https://user-images.githubusercontent.com/25272822/45453076-02c20780-b6e0-11e8-9c4f-52d968b2365f.png)

So as the previous comment, you may find it easy to read but for others it's somehow different.